tests/backup/backup_kdmp_test.go

```diff
@@ -0,0 +1,722 @@
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/volume/portworx/schedops"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
+)
+
+var _ = Describe("{ExcludeDirectoryFileBackup}", func() {
+	var (
+		backupName           string
+		scheduledAppContexts []*scheduler.Context
+		AppContextsMapping   map[string]*scheduler.Context
+		namespace            string
+		bkpNamespaces        []string
+		backupNames          []string
+		restoreNames         []string
+		scheduleNames        []string
+
+		clusterUid        string
+		clusterStatus     api.ClusterInfo_StatusInfo_Status
+		restoreName       string
+		cloudCredName     string
+		cloudCredUID      string
+		backupLocationUID string
+		bkpLocationName   string
+
+		numDeployments int
+
+		providers         []string
+		backupLocationMap map[string]string
+		labelSelectors    map[string]string
+
+		storageClassExcludeFileDirMap map[string][]string
+		mountPathExcludeFileDirMap    map[string][]string
+		existingFileDirMountPathMap   map[string][]string
+		podMountPathMap               map[string][]string
+		fileListMountMap              map[string][]string
+		dirListMountMap               map[string][]string
+		masterDirFileList             map[string][]string
+		scMountPathMap                map[string]string
+		backupNamespaceMap            map[string]string
+		formattedFileString           string
+		fileList                      []string
+		dirList                       []string
+		preRuleName                   string
+		postRuleName                  string
+		preRuleUid                    string
+		postRuleUid                   string
+		periodicSchedulePolicyName    string
+		periodicSchedulePolicyUid     string
+		mutex                         sync.Mutex
+		restoredNamespaces            []string
+		wg                            sync.WaitGroup
+	)
+	JustBeforeEach(func() {
+		backupNames = make([]string, 0)
+		restoreNames = make([]string, 0)
+		//scheduleNames = make([]string, 0)
+		bkpNamespaces = make([]string, 0)
+		restoredNamespaces = make([]string, 0)
+		backupLocationMap = make(map[string]string)
+		labelSelectors = make(map[string]string)
+		AppContextsMapping = make(map[string]*scheduler.Context)
+		storageClassExcludeFileDirMap = make(map[string][]string)
+		mountPathExcludeFileDirMap = make(map[string][]string)
+		existingFileDirMountPathMap = make(map[string][]string)
+		podMountPathMap = make(map[string][]string)
+		fileListMountMap = make(map[string][]string)
+		masterDirFileList = make(map[string][]string)
+		scheduleNames = make([]string, 0)
+		dirListMountMap = make(map[string][]string)
+		scMountPathMap = make(map[string]string)
+		backupNamespaceMap = make(map[string]string)
+		numDeployments = 1
+		providers = getProviders()
+
+		StartPxBackupTorpedoTest("ExcludeDirectoryFileBackup", "Excludes mentioned directories or files from backed-up apps and restores them", nil, 0, Ak, Q4FY24)
```

Comment:
Please update the testcase id https://portworx.testrail.net/index.php?/cases/view/93691

Reply:
Updated